usaspending_api/etl/transaction_loaders/field_mappings_fpds.py

```diff
@@ -542,3 +542,21 @@
     "create_date": current_datetime,  # Data loader won't add this value if it's an update
     "update_date": current_datetime,
 }
+
+
+def all_broker_columns():
+    retval = []
+    retval.extend(transaction_fpds_columns.keys())
+    retval.extend(transaction_fpds_boolean_columns.keys())
+    retval.extend(transaction_normalized_columns.keys())
```

Comment:
100% of the columns from Broker should be landing in `transaction_fpds`.
Including the mappings for the other tables is confusing. Obvious the `set()` below removes duplicates but the usefulness of adding all the mappers is questionable

Reply:
The usefulness is to explicitly add all dependencies. A quick count query seems to indicate that the two tables have the same number of columns, but ignoring the other columns now adds another implicit promise that they will stay that way forever, and adds a THIRD place this loader code could break on the same change.

Reply:
> The usefulness is to explicitly add all dependencies.

Which is accomplished by using the two FPDS table mappers: `transaction_fpds_columns` and `transaction_fpds_boolean_columns`.

> ... but ignoring the other columns now adds another implicit promise that they will stay that way forever, ...

No, it just means that if a future developer wants to have broker values skip `transaction_fpds `and go straight into a "normalized" table they would need to add the mapping to the list, which would be a red flag for the code reviewers.

> ... adds a THIRD place this loader code could break on the same change.

How does that logically follow?  The understanding I held is that the mapper dictionary(ies) for `transaction_fpds`  *should* define 100% of the desired Broker columns. Define once, almost like a specification, and reuse when appropriate. If all of the columns are defined in a single location why would a modification for adding a new column be so confusing?

Reply:
I would advise polling other developers as to if they had this understanding, because it certainly wasn't clear to me until you spelled it out in these reviews. Just looking at our current structure, I would have advised that any new column skip ahead and go straight into the table it's being used at, rather than be stored in an extra table. If we really want to enforce this idea that all tables need to be duplicating all values from the transaction_fpds table, then we should probably change our mapping code to pull from fpds instead of directly from broker, rather than trying to ensure that constraint via the columns we pull from broker.

Reply:
It should be common knowledge that {{transaction_fpds}} is almost a complete copy of Broker's {{detached_award_procurement}}. The differences are minor, and a better model architecture would have the differences as close to 0 as possible (no alterations or subtractions of the data, if any additions they would be very minor)

The ETL loading using the "Broker" data to populate {{transaction_fpds}} and the "normalized" and derived tables is a matter of convenience. It would be more pure to load {{transaction_fpds}} first then load other tables using that data, but it doesn't change the end result.

{{transaction_fpds}} isn't an extra table, it's the source of truth for FPDS data in USAspending. It is also heavily used, a majority of the columns are fetched when creating download CSVs.